101-datastore-create-blob/azuredeploy.json

```diff
@@ -0,0 +1,110 @@
+{
+	"$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+	"contentVersion": "1.0.0.0",
+	"parameters": {
+		"workspaceName": {
+			"type": "string",
+			"metadata": {
+				"description": "Specifies the name of the Azure Machine Learning workspace which will hold this datastore target."
+			}
+		},
+		"datastoreName": {
+			"type": "string",
+			"metadata": {
+				"description": "The name of the datastore, case insensitive, can only contain alphanumeric characters and underscore"
+			}
+		},
+		"accountName": {
+			"type": "string",
+			"metadata": {
+				"description": "The name of the storage account."
+			}
+		},
+		"containerName": {
+			"type": "string",
+			"metadata": {
+				"description": "The name of the azure blob container."
+			}
+		},
+		"storageAccountSubscriptionId": {
+			"type": "string",
+			"defaultValue":"",
+			"metadata": {
+				"description": "Optional : The subscription ID of the storage account."
+			}
+		},
+		"storageAccountResourceGroup": {
+			"type": "string",
+			"defaultValue":"",
+			"metadata": {
+				"description": "Optional : The resource group of the storage account."
+			}
+		},
+		"accountKey": {
+			"type": "securestring",
+			"defaultValue":"",
+			"metadata": {
+				"description": "Optional : A storage account key, defaults to None. Skip this if SAS token authentication is used."
+			}
+		},
+		"sasToken": {
+			"type": "securestring",
+			"defaultValue":"",
+			"metadata": {
+				"description": "Optional : An account SAS token, defaults to None. Skip this if storage account key authentication is used."
```

Comment:
what happens if both are provided or neither is provided?

Reply:
For both scenerio:  We ignore account key and only consider SAS token. Updated the metadata for "storageAcountKey" to reflect this.

For neither scenerio: We create credential-less datastore object. This is a valid customer scenario for datastore object.

Reply:
I think we need to split the params for this:
1) to specify the authn type (none, sas, key)
2) one to contain the secureString (sasToken or Key)

And then use appropriate logic on the resource to set the properties as appropriate

Reply:
ok going to change

Reply:
Fixed this